packages/roosterjs-content-model-plugins/lib/tableEdit/editors/features/TableMover.ts

```diff
@@ -129,3 +157,212 @@ function isTableTopVisible(editor: IEditor, rect: Rect | null, contentDiv?: Node
 
     return true;
 }
+
+function setTableMoverCursor(editor: IEditor, state: boolean, type?: 'move' | 'copy') {
+    editor?.setEditorStyle(TABLE_MOVER_STYLE_KEY, state ? 'cursor: ' + type ?? 'move' : null);
+}
+
+function onDragStart(context: TableMoverContext, event: MouseEvent) {
+    context.onStart();
+
+    const { editor, table, div } = context;
+
+    setTableMoverCursor(editor, true, 'move');
+
+    // Create table outline rectangle
+    const trect = table.getBoundingClientRect();
+    const createElementData = {
+        tag: 'div',
+        style: 'position: fixed; user-select: none; border: 1px solid #808080',
+    };
+    const tableRect = createElement(createElementData, document) as HTMLDivElement;
+    tableRect.style.width = `${trect.width}px`;
+    tableRect.style.height = `${trect.height}px`;
+    tableRect.style.top = `${trect.top}px`;
+    tableRect.style.left = `${trect.left}px`;
+    div.parentNode?.appendChild(tableRect);
+
+    // Get current selection
+    const initialSelection = editor.getDOMSelection();
+
+    // Select first cell of the table
+    editor.setDOMSelection({
+        type: 'table',
+        firstColumn: 0,
+        firstRow: 0,
+        lastColumn: 0,
+        lastRow: 0,
+        table: table,
+    });
+
+    // Get the table content model
+    const [cmTable] = getFirstSelectedTable(editor.getContentModelCopy('disconnected'));
+
+    // Restore selection
+    editor.setDOMSelection(initialSelection);
```

Comment:
Instead of set selection then move it back, you can do:

```ts
let cmTable: ContentModelTable|undefined;

editor.formatContentModel(model => {
  cmTable = getFirstSelectedTable(model);
  return false;
}, {
  selectionOverride: { type: 'table',... }
});
```

This allows you use a different selection to create content model and not impact the original one. So no need to set selection away then set back.

Reply:
Seeing as this method is implemented to obtain the cmTable on all the other Table plugins, I prefer to make this optimisation on a separate PR for all plugins.

Reply:
Here is that PR #2656